### DISCUSSION TITLE
PVE Helper Scripts taking a LONG time to complete (20-30-40 minutes) - Any debugging advice?

### DISCUSSION THREAD

**Discussion by DerekGreen** [tteck/Proxmox#3090] [Help Needed] [answered]
Firstly, a big thanks to @tteck for the wonderfully helpful scripts 👍🙂

When I run the scripts on my recently installed PVE 8.2.2 server, they are taking a LONG time to complete, sometimes 20, 30, 40 minutes (which I'm presuming is not normal 🤔)

For example, I just ran the Docker (Default) script and it took 33 mins (including a couple of minutes to set Advanced options)
 - It sat on "Creating a Docker LXC using the above advanced settings" for around 8-9 mins
 - "Updating LXC Template List" took over 12 mins (despite the 'Debian 12 standard' template already downloaded)
 - The next 2 lines "LXC Container X was successfully created" and "Started LXC Container" were almost instant... but then it sat there showing nothing for 8-9 minutes (before I saw all the installation commands scrolling quickly past)
  - All the final commands completed in a minute or so (including when I had to answer the questions about Portainer and Compose)

I have the 'verbose' script option enabled, so I am seeing the output in the console window! (Nothing else of note, apart from the long times between task-lines)

It is a small mini PC (N100 16GB-DDR4 512GB-nvme), currently running a pfSense VM and 1-2 small LXCs, and is showing around 3-5% CPU usage and 6 of 16GB memory used, throughout the script execution!

The 'Post Install' script runs through quickly (but obviously that has nothing to do with LXCs)

Manually creating a CT in PVE (using the previously downloaded Debian 12 std template) took around 20-30 seconds!

N.B. I was previously having a problem where the scripts would do 'nothing' after running, but I fixed this by adding the -4 option after wget (so there perhaps is some sort of ip6 issue - I do specify Disable ip6 when running the Helper Scripts)

I have searched the 'Discussions' page for "slow" + "long" and I couldn't see anything obviously similar.

I'm guessing some sort of long 'timeouts' are happening 🤷
 - Any suggestions for other places to look or other ways to debug this issue??
 
Not a major issue, but I'd very much like to work out what is going on.
Thanks in advance to anyone who can offer any advice!

Cheers, D.

**Comment by tteck** [comment]
**Your situation is NOT normal.**
A script is a set of instructions for your computer to follow. Slow download or execution times could indicate network issues or a lack of resources. If I had to guess, the issue might lie with the "pfSense VM".

**Comment by jmaldon1** [comment, marked as answer]
My scripts were taking a long time and then failing, it turns out it was because the DNS IP was defaulting to the host IP but I needed it to be a different IP. Scripts starting working immedietly after I changed this in the advanced set up options.

**Reply by jonnywright** [reply]
I had this exact issue just hours ago. Brand new PVE8 install had incorrect DNS IP set.

**Reply by jpmurray** [reply]
This is a lifesaver ! I've spent my whole day trying to install Plex via the scripts, now putting a correct DNS IP it zipped through install in a matter of minutes!
